portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,25 +38,33 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    PreparedQuery results = datastore.prepare(new Query("Comment"));
-    List<String> comments = new ArrayList<>();
+    PreparedQuery results =
+        datastore.prepare(new Query("Comment").addSort("timestamp", SortDirection.DESCENDING));
+    List<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
       String message = (String) entity.getProperty("message");
-      comments.add(message);
+      long timestamp = (long) entity.getProperty("timestamp");
+      comments.add(new Comment(id, name, timestamp, message));
     }
     response.setContentType("text/html;");
     response.getWriter().println(new Gson().toJson(comments));
   }
 
   // Puts inputted comments into database and redirects result back to page
-  // TO DO: check if input is null
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", getParameter(request, "name-input", "Anonymous"));
     commentEntity.setProperty("message", getParameter(request, "text-input", /* defaultValue=*/""));
+    commentEntity.setProperty("timestamp", System.currentTimeMillis());
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+
+    if (!((String) commentEntity.getProperty("message")).isEmpty()) {
+      datastore.put(commentEntity);
+    }
```

Comment:
Seems this can be in a private method, e.g. wrtiteToDatastore(request);

Reply:
for the future , when is it good to make a method in similar cases (where I am not reusing code, just using once)? or is it just to chunk the code better?

Reply:
done